src/Botan.php

```diff
@@ -181,33 +227,37 @@ public static function shortenUrl($url, $user_id)
         }
 
         $cached = BotanDB::selectShortUrl($user_id, $url);
-        if (!empty($cached[0]['short_url'])) {
-            return $cached[0]['short_url'];
-        }
 
-        $request = str_replace(
-            ['#TOKEN', '#UID', '#URL'],
-            [self::$token, $user_id, urlencode($url)],
-            self::$shortener_url
-        );
+        if (!empty($cached)) {
```

Comment:
`!empty()` is redundant here, as `$cached` gets defined above.
You could also just do:
```php
if ($cached = BotanDB::selectShortUrl($user_id, $url)) {
    return $cached;
}
```